tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -1938,4 +1967,139 @@ public function userHasRestoredSpaceRequest(
 			"Expected response status code should be $expectedHTTPStatus"
 		);
 	}
+
+	/**
+	 * @When /^user "([^"]*)" lists all deleted files in the trash bin of the space "([^"]*)"$/
+	 *
+	 * @param  string $user
+	 * @param  string $spaceName
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userListAllDeletedFilesinTrash(
+		string $user,
+		string $spaceName
+	): void {
+		$space = $this->getSpaceByName($user, $spaceName);
+		$fullUrl = $this->baseUrl . "/remote.php/dav/spaces/trash-bin/" . $space["id"];
+		$this->featureContext->setResponse(
+			HttpRequestHelper::sendRequest(
+				$fullUrl,
+				"",
+				'PROPFIND',
+				$user,
+				$this->featureContext->getPasswordForUser($user),
+				[],
+				""
+			)
+		);
+	}
+
+	/**
+	 * User get all objects in the trash of project space 
+	 * 
+	 * method "getTrashbinContentFromResponseXml" borrowed from core repository
+	 * and return array like:
+	 * 	[1] => Array
+	 *       (
+	 *             [href] => /remote.php/dav/spaces/trash-bin/spaceId/objectId/
+	 *             [name] => deleted folder
+	 *             [mtime] => 1649147272
+	 *             [original-location] => deleted folder
+	 *        )
+	 *
+	 * @param  string $user
+	 * @param  string $spaceName
+	 *
+	 * @return array
+	 * @throws GuzzleException
+	 */
+	public function getObjectsInTrashbin(
+		string $user,
+		string $spaceName
+	): array {
+		$this->userListAllDeletedFilesinTrash($user, $spaceName);
+		$this->featureContext->theHTTPStatusCodeShouldBe(
+			207,
+			"Expected response status code should be 207"
+		);
+		return $this->trashbinContext->getTrashbinContentFromResponseXml(
+			$this->featureContext->getResponseXml($this->featureContext->getResponse())
+		);
+	}
+
+	/**
+	 * @Then /^as "([^"]*)" (?:file|folder|entry) "([^"]*)" should (not|)\s?exist in the trashbin of the space "([^"]*)"$/
+	 *
+	 * @param  string $user
+	 * @param  string $object
+	 * @param  string $shouldOrNot   (not|)
+	 * @param  string $spaceName
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function listObjectsInTrashbin(
```

Comment:
I find the function name a bit confusing, maybe change it to something like:

```suggestion
	public function checkExistanceOfObjectsInTrashbin(
```